### DISCUSSION TITLE
I dont know why when i want to use "Address Lookup Table" OR "Creating new account by system program" i ran into unexpected error!

### DISCUSSION THREAD

**Discussion by pooriagg** [phantom/docs#136] [Help & Support] [answered]
### Summary

I dont know why when i want to use "Address Lookup Table" OR "Creating new account by system program" i ran into unexpected error!

This is my react code that runs into unexpected error after approving transaction in phantom :

(I am using phantom in developer mode - devnet)

### Steps to Reproduce

 -Sending Create Account Instruction But getting "Unexpected error from phantom" !

    const createPDAAccount = useCallback(async () => {
      // try {
        if (!userPubKey) {
          throw new Error("User not found!");
        };
      const user = new solanaJs.PublicKey(userPubKey);

      const devnet = new solanaJs.Connection(
        solanaJs.clusterApiUrl("devnet"),
        "confirmed"
      );
      const programId = new solanaJs.PublicKey(
        "BHY5jmsWXDAWzbjUaiwAZGcjyfMYNNYgk4ooESgCPWvL"
      );
      const size = 42; // Bytes
      const rent = await devnet.getMinimumBalanceForRentExemption(size);
      const [ pda ] = solanaJs.PublicKey.findProgramAddressSync(
        [
          Buffer.from("account"),
          user.toBuffer()
        ],
        programId
      );

      const ix = solanaJs.SystemProgram.createAccount(
        {
          lamports: rent,
          fromPubkey: user,
          newAccountPubkey: pda,
          space: size,
          programId
        }
      );

      const txMsg = new solanaJs.TransactionMessage(
        {
          payerKey: user,
          recentBlockhash: (await devnet.getLatestBlockhash()).blockhash,
          instructions: [ ix ]
        }
      ).compileToV0Message();
      const tx = new solanaJs.VersionedTransaction(txMsg);

          const { signature: txSig } = await phantom.signAndSendTransaction(tx);
          console.log(`Transaction signature - ${txSig}`);
        // } catch {
        //   console.error("Failed create program owned pda account.");
        // };
      }, [ userPubKey, phantom ]);

-I Create and Extend a LUT Table BUT When I Want To Send A Transaction Using It I Am getting "Unexpected error from phantom" !

    const sendALT = useCallback(async () => {
      const user = new solanaJs.PublicKey(userPubKey!);
      const latestBlock = await solana.getLatestBlockhash();
  
      const [ altInstruction, altPubKey ] = solanaJs.AddressLookupTableProgram.createLookupTable({
        payer: user,
        authority: user,
        recentSlot: await solana.getSlot()
      });

    const txMsg = new solanaJs.TransactionMessage({
      payerKey: user,
      recentBlockhash: latestBlock.blockhash,
      instructions: [ altInstruction ]
    }).compileToV0Message();
    const tx = new solanaJs.VersionedTransaction(txMsg);
    const { signature: txSig } = await phantom.signAndSendTransaction(tx);
    await solana.confirmTransaction({
      signature: txSig,
      blockhash: latestBlock.blockhash,
      lastValidBlockHeight: latestBlock.lastValidBlockHeight
    }, "finalized");
    console.log(`Transaction signature - ${txSig}.`);
    console.log(`Address lookup Table - ${altPubKey}.`);

    const extendInstruction = solanaJs.AddressLookupTableProgram.extendLookupTable({
      payer: user,
      authority: user,
      lookupTable: altPubKey,
      addresses: [
        user,
        solanaJs.SystemProgram.programId
      ]
    });
    const txMsg_2 = new solanaJs.TransactionMessage({
      payerKey: user,
      recentBlockhash: latestBlock.blockhash,
      instructions: [ extendInstruction ]
    }).compileToV0Message();
    const tx_2 = new solanaJs.VersionedTransaction(txMsg_2);
    const { signature: txSig_2 } = await phantom.signAndSendTransaction(tx_2);
    await solana.confirmTransaction({
      signature: txSig_2,
      blockhash: latestBlock.blockhash,
      lastValidBlockHeight: latestBlock.lastValidBlockHeight
    }, "finalized");
    console.log("Table extended.");
    console.log(`Transaction signature - ${txSig}.`);
  
    const instruction: solanaJs.TransactionInstruction = solanaJs.SystemProgram.transfer({
      fromPubkey: user,
      toPubkey: solanaJs.Keypair.generate().publicKey,
      lamports: 1e6
    });

    const ATL = await solana.getAddressLookupTable(altPubKey).then(res => res.value);

    const transactionMsg = new solanaJs.TransactionMessage({
      payerKey: user,
      recentBlockhash: latestBlock.blockhash,
      instructions: [ instruction ]
    }).compileToV0Message([ ATL! ]);
    const transaction = new solanaJs.VersionedTransaction(transactionMsg);

    const { signature: txSig_3 } = await phantom.signAndSendTransaction(transaction); //!

    await solana.confirmTransaction({
      signature: txSig_3,
      blockhash: latestBlock.blockhash,
      lastValidBlockHeight: latestBlock.lastValidBlockHeight
    }, "finalized");
    
    const { value } = await solana.getSignatureStatus(txSig);
      console.log("Tx result -", value);
    }, [ userPubKey, solana, phantom ]);

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment, marked as answer]
Hi @pooriagg,

Sorry for the delay in response! Have you taken a look at our sandbox implementation here: https://github.com/0xproflupin/phantom-sandboxes/blob/main/src/sandboxes/sol-sandbox/utils/signAndSendTransactionV0WithLookupTable.ts
